tests/test_selector.py

```diff
@@ -724,6 +747,28 @@ def test_re_intl(self) -> None:
         x = self.sscls(text=body)
         self.assertEqual(x.xpath("//div").re(r"Evento: (\w+)"), ["cumplea\xf1os"])
 
+    def test_re_flags(self) -> None:
+        body = """
+        <script>
+            function example() {
+            "name": "Adrian",
+            "points": 3,
+            }
+        </script>
+        """
+        sel = self.sscls(text=body)
+
+        self.assertEqual(sel.xpath("//script/text()").re(r"example\(\) ({.*})"), [])
+        self.assertEqual(
+            sel.xpath("//script/text()").re(r"example\(\) ({.*})", flags=re.DOTALL),
+            [
+                """{
+            "name": "Adrian",
+            "points": 3,
+            }"""
+            ],
+        )
+
```

Comment:
We can add more examples like with `re.I` (ignore case sensitive), but I'm not sure if it's necessary...

Reply:
I agree, not necessary. 1 flag should be enough, otherwise we would be actually testing the Python standard library more than our own code :)